libsolidity/ast/Types.cpp

```diff
@@ -439,6 +439,83 @@ MemberList::MemberMap Type::boundFunctions(Type const& _type, ContractDefinition
 	return members;
 }
 
+AddressType::AddressType()
+{
+}
+
+string AddressType::richIdentifier() const
+{
+	return "t_address";
+}
+
+bool AddressType::isImplicitlyConvertibleTo(Type const& _convertTo) const
+{
+	return _convertTo.category() == category();
+}
+
+bool AddressType::isExplicitlyConvertibleTo(Type const& _convertTo) const
+{
+	return isImplicitlyConvertibleTo(_convertTo) ||
+		   _convertTo.category() == Category::Contract ||
+		   _convertTo.category() == Category::Integer ||
+		   _convertTo.category() == Category::Enum ||
+		   (_convertTo.category() == Category::FixedBytes && 160 == dynamic_cast<FixedBytesType const&>(_convertTo).numBytes() * 8) ||
+		   _convertTo.category() == Category::FixedPoint;
+}
+
+bool AddressType::operator==(Type const& _other) const
+{
+	return _other.category() == category();
+}
+
+string AddressType::toString(bool) const
+{
+	return "address";
+}
+
+u256 AddressType::literalValue(Literal const* _literal) const
+{
+	solAssert(_literal, "");
+	solAssert(_literal->value().substr(0, 2) == "0x", "");
+	return u256(_literal->value());
+}
+
+TypePointer AddressType::unaryOperatorResult(Token::Value _operator) const
+{
+	return _operator == Token::Delete ? make_shared<TupleType>() : TypePointer();
+}
+
+
+TypePointer AddressType::binaryOperatorResult(Token::Value _operator, TypePointer const& _other) const
+{
+	if (
+		_other->category() != Category::RationalNumber &&
+		_other->category() != Category::FixedPoint &&
```

Comment:
I think this line should be removed.

Reply:
Actually the whole `if` check here is redundant due to the call to `commonType` below, isn't it?